### PR TITLE
Task-06: implement routing logic

### PR DESCRIPTION
**Pull request by jaxin007**
Acceptance Criteria for 1
- [x]  Mentor is added to pr
- [x]  AC copied to pr description
- [x]  Link to deployed version is added to pr
- [x]  LayoutComponent with a static header and footer, and an Outlet for dynamic content, is added.
- [x]  Two routes, AboutPage and ProductsPage, are added.
- [x]  The AboutPage is set as the index route.
- [x]  A route for invalid pages is added, displaying Page not found with a link to the AboutPage.

Acceptance Criteria for 1*
- [ ]  The layout for ProductPage is added according to the design.
- [ ]  An additional route for ProductPage with dynamic parameters :id is added.
- [ ]  An additional request getProduct is implemented, using the product's id.

### CODE REVIEW

File: src/components/Header/Header.module.css [hunk 1]
```diff
@@ -53,6 +53,8 @@
 .activeButton {
   background: none;
   font-weight: 700 !important;
```
Comment (tetiana-zhuchenko) [inline src/components/Header/Header.module.css]: Could you find out how to implement `bold` style to your active page without using `!important`. It's a bad practice to use.

File: src/components/Header/Header.module.css [hunk 2]
```diff
@@ -53,6 +53,8 @@
...
+  display: flex;
+  align-items: center;
 }
 
 .inactiveButton {
```
Comment (tetiana-zhuchenko) [inline src/components/Header/Header.module.css:60]: Are you sure that you need to use `nesting` there?

File: src/components/Header/Header.component.tsx [hunk 3]
```diff
@@ -14,48 +13,31 @@ import type { Themes } from '@/types/themes.type.ts';
...
     theme: Themes;
     setTheme: (theme: Themes) => void;
 }
 
-export const Header: React.FC<HeaderProps> = ({ totalProductsInCart, handleActivePage, activePage, theme, setTheme }) => (
+export const Header: React.FC<HeaderProps> = ({ totalProductsInCart, handleActivePage, theme, setTheme }) => (
     <div className={styles.headerWrapper}>
         <div className={styles.header}>
             <div className={styles.leftIcons}>
                 <IconWithRedirect icon={images.maCrownIcon} url={links.home} alt={'masters academy icon'}></IconWithRedirect>
```
Comment (tetiana-zhuchenko) [inline src/components/Header/Header.component.tsx:26]: I hope that this link should lead to "Masters Academy" website, but it's up to you. Later you may add your logo there…

File: src/pages/NotFound.page.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+export const NotFoundPage = () => {
+    const navigate = useNavigate();
+
+    return (
+        <>
+            <div className={styles.notFoundWrapper}>
+                <div className={styles.notFoundContent}>
+                    <h1 className={styles.numbers}>404</h1>
+                    <h3 className={styles.notFoundText}>Page not found</h3>
+                    <button className={styles.backHomeButton} onClick={() => navigate('/')}>
```
Comment (tetiana-zhuchenko) [inline src/pages/NotFound.page.tsx]: Had you considered using `NavLink` there?

File: src/pages/NotFound.page.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { useNavigate } from 'react-router-dom';
+
+import styles from './NotFound.module.css';
+
+export const NotFoundPage = () => {
+    const navigate = useNavigate();
```
Comment (tetiana-zhuchenko) [inline src/pages/NotFound.page.tsx]: When you will use `NavLink` bellow, you don't need any `hooks` there

File: src/App.tsx [hunk 6]
```diff
@@ -49,16 +44,16 @@ function App() {
...
-                {activePage === 'about' && <AboutMe />}
-                {activePage === 'products' && <ProductWrapper theme={currentTheme} updateTotalCart={updateTotalCart} />}
-                <Footer />
+                <Routes>
+                    <Route
+                        path={'/'}
+                        element={<LayoutComponent totalProductsInCart={totalProductsInCart} theme={currentTheme} setTheme={handleTheme} />}
+                    >
+                        <Route index element={<AboutMe />} />
+                        <Route path={'/products'} element={<ProductWrapper theme={currentTheme} updateTotalCart={updateTotalCart} />} />
```
Comment (tetiana-zhuchenko) [inline src/App.tsx:53]: Do you have any plans to implement `Dynamic Route for ProductPage` to accomplish `Advance Task`

**Review by tetiana-zhuchenko — COMMENTED** [review]
Dima, thanks for your job! Your app works great, and your code is pretty good, and l recommend you to make some tiny improvements.

File: src/App.tsx (hunk 6 above)
Comment (jaxin007) [reply src/App.tsx:53]: I was hoping to finish that task asap and take care of Product Page Later

File: src/pages/NotFound/NotFound.page.tsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+import styles from './NotFound.module.css';
+
+export const NotFoundPage = () => (
+    <>
+        <div className={styles.notFoundWrapper}>
+            <div className={styles.notFoundContent}>
+                <h1 className={styles.numbers}>404</h1>
+                <h3 className={styles.notFoundText}>Page not found</h3>
+                <NavLink to="/">
```
Comment (tetiana-zhuchenko) [inline src/pages/NotFound/NotFound.page.tsx:11]: I'm sorry, I've made the mistake there. The correct code should look like this:
`<Link to="/">Go back home</Link>` as you see, you don't need `<button>` there
https://reactrouter.com/en/main/components/link

**Review by tetiana-zhuchenko — APPROVED** [review]
Well done! Thanks for your job!

**Pull request merged**